### ISSUE TITLE
[BUG] Set-PnPListItem not clearing taxonomy field

### ISSUE THREAD

**Issue by randhawaryan** [pnp/powershell#2830]
### Reporting an Issue or Missing Feature
I created a new list TestPnPList
Created two multi-value managed metadata fields in the list: MMSField1, MMSField2
Created a single line of text field TextField
Manually created and populated a few test list items (populating each field)

When running a command to update MMSField1 and clear MMSField2, MMSField2 is not cleared. 
If the script is updated to only clear MMSField2, the field is cleared. 

Connect-PnPOnline -Url $url -Interactive
$terms = @("99aef3e7-5077-4a79-b740-e770a805af8a", "c4c5f009-583e-4be4-903a-e8369f8a4cc3")
Set-PnPListItem -List "TestPnPList" -Identity 1 -Values @{"MMSField1"=($terms); "MMSField2"= $null; "TextField"="Updated by script"} -UpdateType SystemUpdate

### Expected behavior 
MMSField1 populated with two values, TextField populated.
MMSField2 cleared.

Script output: 
Id    Title                                              GUID                                                                                                                            
--    -----                                              ----                                                                                                                            
1     TestItem                                           3462dc53-520e-410c-a539-0279860f5073                                                                                            

### Actual behavior 
MMSField1 and TextField are populated. 
MMSField2 is not cleared (old value remains)

### Steps to reproduce behavior
Connect-PnPOnline -Url $url -Interactive
$terms = @("99aef3e7-5077-4a79-b740-e770a805af8a", "c4c5f009-583e-4be4-903a-e8369f8a4cc3")
Set-PnPListItem -List "TestPnPList" -Identity 1 -Values @{"MMSField1"=($terms); "MMSField2"= $null; "TextField"="Updated by script"} -UpdateType SystemUpdate

The following line of code does clear MMSField2 (and set TextField)
Set-PnPListItem -List "TestPnPList" -Identity 2 -Values @{"MMSField2"= $null; "TextField"="Updated by script1"} -UpdateType SystemUpdate

### What is the version of the Cmdlet module you are running?
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)
1.12.0

### Which operating system/environment are you running PnP PowerShell on?
- [ X] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by KoenZomers** [comment]
@randhawaryan I cannot reproduce this. It seems to work well here.

My before scenario:
<img width="693" alt="image" src="https://user-images.githubusercontent.com/5940670/220649186-33bf161d-022d-450b-bb5d-cc9bad12694c.png">

The lines I run:
```powershell
$terms = @("5b03eda9-4f80-472f-b20a-bf5517238106", "6b82c553-edef-47f6-9456-9d0c1732eaf9")
Set-PnPListItem -List "Issue2830" -Identity 1 -Values @{"MMSField1"=($terms); "MMSField2"= $null; "TextField"="Updated by script"} -UpdateType SystemUpdate
```

The result:
<img width="695" alt="image" src="https://user-images.githubusercontent.com/5940670/220649506-f9fd3804-ec3a-45e0-add4-941426cc7031.png">

Am I doing something different? Can you try updating to the latest nightly build using `Update-Module PnP.PowerShell -AllowPrerelease` and test it with that version in PowerShell 7.2 or later to see if it works?

**Comment by randhawaryan** [comment]
Hi @KoenZomers
I'm able to get the same behaviour as you if I use PowerShell 7.2 + nightly build, however it only works if I add the following line to the script:
Import-Module PnP.PowerShell -MinimumVersion 2.0.0

I'm assuming the reasoning behind this is out of scope - I think we can close this ticket. 

Some additional details: 
This may be because of a version mismatch on my side (I tried uninstalling all versions and installing only the nightly build). 
After uninstalling the nightly build and installing only version 1.12.0, I can get this script to work if I use
Import-Module PnP.PowerShell -MinimumVersion 1.12.0
Oddly enough, if I run the same thing with Import-Module PnP.PowerShell -RequiredVersion 1.12.0 I see my original issue. 

When trying the above steps - Get-Module PnP.PowerShell -ListAvailable is only returning one version (either nightly build or 1.12.0) since I uninstalled all versions and re-installed the versions I was testing with.
